src/TextField/TextFieldLabel.js

```diff
@@ -35,13 +43,30 @@ export default function TextFieldLabel(props, context) {
   }, classNameProp);
 
   return (
-    <label className={className} {...other} />
+    <label className={className} {...other}>
+      {children}
+      {required ? <span className={asteriskClassName}>{'\u2009'}*</span> : null}
```

Comment:
Looks much cleaner, I'll change that! :+1: Damn... that'd make a lot of code I wrote in the last weeks a little more beautiful. :smile: